code/obj/item/mops_cleaners.dm

```diff
@@ -1192,3 +1194,99 @@ TYPEINFO(/obj/item/handheld_vacuum/overcharged)
 		animate(alpha=0, time=0.21 SECONDS, easing=SINE_EASING)
 		SPAWN(0.5 SECONDS)
 			qdel(src)
+
+
+/obj/item/trash_bag
+	name = "trash bag"
+	desc = "A flimsy bag for filling with things that are no longer wanted."
+	icon = 'icons/obj/janitor.dmi'
+	inhand_image_icon = 'icons/mob/inhand/hand_tools.dmi'
+	icon_state = "trashbag-f"
+	item_state = "trashbag-f"
+	uses_multiple_icon_states = 1
+	w_class = W_CLASS_TINY
+	rand_pos = 1
+	flags = FPRINT | TABLEPASS | NOSPLASH
+	tooltip_flags = REBUILD_DIST
+	var/base_state = "trashbag"
+
+	New()
+		..()
+		src.create_storage(/datum/storage/no_hud, prevent_holding = list(/obj/item/trash_bag), max_wclass = W_CLASS_NORMAL, slots = 20,
+			params = list("use_inventory_counter" = TRUE, "variable_weight" = TRUE, "max_weight" = 20))
+
+	equipped(mob/user)
+		..()
+		for (var/i = 1 to round(length(src.storage.get_contents()) / 3))
+			src.remove_random_item(user)
+
+	attackby(obj/item/I, mob/user)
+		if (issnippingtool(I))
+			var/list/actions = list("Cut into an outfit", "Put it into the bag")
+			var/action = input(user, "What do you want to do with [src]?") as null|anything in actions
+			if (!action)
+				return
+			switch (action)
+				if ("Cut into an outfit")
+					boutput(user, "You begin cutting up [src].")
+					if (!do_after(user, 3 SECONDS))
+						boutput(user, "<span class='alert'>You were interrupted!</span>")
+						return
+					else
+						var/obj/item/clothing/under/shorts/trashsinglet/trash_outfit = new /obj/item/clothing/under/shorts/trashsinglet(get_turf(src))
+						playsound(src.loc, 'sound/items/Scissor.ogg', 100, 1)
+
+						user.u_equip(src)
+						if (length(src.storage.get_contents()))
+							for(var/obj/item/contents in src.storage.get_contents())
+								src.storage.transfer_stored_item(contents, get_turf(src))
+							boutput(user, "You cut leg holes into [src] and all the contents fall out!")
+						else
+							boutput(user, "You cut leg holes into [src].")
+						usr.put_in_hand_or_drop(trash_outfit)
+						qdel(src)
+						return
+		..()
+
+	attack_hand(mob/user)
+		..()
+		var/mob/living/carbon/human/H = src.loc
+		if (istype(H) )
+			src.remove_random_item(H)
+
```

Comment:
```suggestion
```
Used for when it was worn, otherwise this will make two items come out every time you click it with empty hand